packages/node_modules/@webex/plugin-memberships/src/memberships.js

```diff
@@ -321,6 +323,89 @@ const Memberships = SparkPlugin.extend({
       .then((res) => new Page(res, this.spark));
   },
 
+  /**
+   * Returns a list of memberships with details about the lastSeenId for each
+   * user, allowing a client to indicate "read status" in a space GUI
+   *
+   * This differs from the memberships.list() function in the following ways:
+   *  -- it accepts only a room or object with a valid roomId
+   *  -- no other options, eg: max, are considered
+   *  -- results are not paginated
+   *  -- memberships in the return do not include the
+   *     "created", "isRoomHidden", fields
+   *  -- memberships in the return do include the new
+   *    "lastSeenId", and "lastSeenDate" fields
+   *     these will not exist if the member has never "seen" the space
+   *
+   * In general this function should be used only when the
+   * client needs to access read status info.
+   *
+   * This function may be deprecated when this info is provided in the membership
+   * objects returned in the list function.
+   *
+   *
+   * @instance
+   * @memberof Memberships
+   * @param {Object} options
+   * @param {string} options.roomId
+   * @returns {Promise<MembershipObjectList>}
+   */
+  listWithReadStatus(options) {
+    const conversation = {
+      id: deconstructHydraId(options.roomId).id
+    };
+
+    return this.spark.internal.conversation.get(conversation,
+      {
+        participantAckFilter: 'all', // show lastAck info for each participant
+        activitiesLimit: 0 // don't send the whole history of activity
+      })
+      .then((resp) => {
+        try {
+          // We keep track of the last read message by each user
+          const roomUUID = resp.id;
+          const roomId = constructHydraId(hydraTypes.ROOM, roomUUID); 
+          const particpants = resp.participants.items;
+          const lastReadInfo = {items: []};
+
+          for (let i = 0; i < particpants.length; i += 1) {
+            const participant = particpants[i];
+            const participantInfo = {
+              id: constructHydraId(hydraTypes.MEMBERSHIP,
+                `${participant.entryUUID}:${roomUUID}`),
+              roomId,
+              personId: constructHydraId(hydraTypes.PEOPLE,
+                participant.entryUUID),
+              personEmail: participant.entryEmailAddress ||
+                participant.entryEmail,
+              personDisplayName: participant.displayName,
+              personOrgId: constructHydraId(hydraTypes.ORGANIZATION,
+                participant.orgId),
+              isModerator: participant.roomProperties.isModerator,
+              isMonitor: false,
+              roomType: resp.tags.includes(SDK_EVENT.INTERNAL.ACTIVITY_TAG.ONE_ON_ONE) ?
+                SDK_EVENT.EXTERNAL.SPACE_TYPE.DIRECT :
+                SDK_EVENT.EXTERNAL.SPACE_TYPE.GROUP
+              // isRoomHidden,
+              // created,
+            };
```

Comment:
This if block is really hard for me to grok. What is your line wrap length set on?

Reply:
Basically I'm converting the conversations participant object into an external membership object.
linter throws a warning for lines longer than 80 so I added new lines in some places, does that ultimately make it harder to understand?

Reply:
Is this any better?